src/main/scala/rules/Executor.scala

```diff
@@ -100,6 +101,30 @@ object executor extends ExecutionRules {
       Q(s, v)
     } else if (edges.length == 1) {
       follow(s, edges.head, v)(Q)
+    } else if (Verifier.config.parallelizeBranches() &&
```

Comment:
Wow, that's a lot of (repeated) casts and typechecks. Wouldn't it be more concise to pattern match on the edge types?